File: frontend/src/features/dashboard/api/dashboard.service.ts
import api from "@/lib/api";
import { AdminStats, AssistantStats } from "@/features/dashboard/types";

export const dashboardService = {
  /**
   * Lấy thống kê cho Admin.
   */
  getStats: async (academicYearId?: number): Promise<AdminStats> => {
    const params = new URLSearchParams();
    if (academicYearId)
      params.append("academicYearId", academicYearId.toString());
    const response = await api.get(
      `/admin/dashboard/stats?${params.toString()}`,
    );
    return response.data;
  },

  /**
   * Lấy thống kê cho Trợ lý khoa.
   */
  getAssistantStats: async (
    departmentId: number,
    academicYearId: number,
  ): Promise<AssistantStats> => {
    const params = new URLSearchParams();
    params.append("departmentId", departmentId.toString());
    params.append("academicYearId", academicYearId.toString());
    const response = await api.get(
      `/assistant/dashboard/stats?${params.toString()}`,
    );
    return response.data;
  },
};

File: backend/src/main/java/vn/edu/husc/researchhub/service/DashboardService.java
package vn.edu.husc.researchhub.service;

import vn.edu.husc.researchhub.dto.response.DashboardStatsResponse;

/**
 * Service quản lý Dashboard.
 * Cung cấp số liệu thống kê cho Admin, Trợ lý, Giảng viên.
 */
public interface DashboardService {
  /**
   * Lấy số liệu thống kê tổng quan (Admin/Assistant).
   */
  DashboardStatsResponse getStats(Integer academicYearId);

  /**
   * Lấy số liệu thống kê dành cho Trợ lý khoa.
   */
  vn.edu.husc.researchhub.dto.response.AssistantDashboardStatsResponse getAssistantStats(
      Integer departmentId, Integer academicYearId);
}
